src/lib/parts.ts

```diff
@@ -37,6 +37,61 @@ export const isIterable = (value: unknown): value is Iterable<unknown> => {
       !!(value && (value as any)[Symbol.iterator]);
 };
 
+/**
+ * A Part that controls all or part of an attribute value.
+ */
+export class SpreadPart implements Part {
+  value: unknown = undefined;
+  subparts: Map<string, ReadonlyArray<Part>> = new Map();
+  element: Element;
+  handleAttribute: (name: string) =>  ReadonlyArray<Part>;
+  // Names of attributes that were spread in the last commit,
+  // but are not anymore.
+  private __removedNames: Array<string> = [];
+
+  constructor(element: Element, handleAttribute: (name: string) =>  ReadonlyArray<Part>) {
+    this.element = element;
+    this.handleAttribute = handleAttribute;
+  }
+
+  setValue(value: unknown): void {
+    let spreadable = value as {[key: string]: any};
+    for (var name in spreadable) {
+      this.subparts.set(name, this.handleAttribute(name))
+    }
+    for (var [name, parts] of this.subparts.entries()) {
+      if (parts) {
+        // Only one part per key is allowed via Spread
+        parts[0].setValue(spreadable[name]);
```

Comment:
should this be in a conditional too?

```js
if (name in spreadable) {
  parts[0].setValue(spreadable[name]);
} else {
  this.__removedNames.push(name);
}
```

Reply:
I purposefully didn't include the `name in spreadable` condition here cause sometimes we want to commit `undefined` to clean up a part. For example, if someone removes an event, committing `undefined` for the EventPart will remove the event listener.